homeassistant/components/sensor/nest.py

```diff
@@ -76,97 +71,58 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
-        """Initialize the sensor."""
-        self.structure = structure
-        self.variable = variable
-
-        if device is not None:
-            # device specific
-            self.device = device
-            self._location = self.device.where
-            self._name = "{} {}".format(self.device.name_long,
-                                        self.variable.replace('_', ' '))
-        else:
-            # structure only
-            self.device = structure
-            self._name = "{} {}".format(self.structure.name,
-                                        self.variable.replace('_', ' '))
-
-        self._state = None
-        self._unit = None
-
-    @property
-    def name(self):
-        """Return the name of the nest, if any."""
-        return self._name
-
-    @property
-    def unit_of_measurement(self):
-        """Return the unit the value is expressed in."""
-        return self._unit
 
-    @property
-    def should_poll(self):
-        """Do not need poll thanks using Nest streaming API."""
-        return False
+    for structure, device in nest.thermostats():
+        all_sensors += [NestBasicSensor(structure, device, variable)
+                        for variable in conditions
+                        if variable in SENSOR_TYPES]
+        all_sensors += [NestTempSensor(structure, device, variable)
+                        for variable in conditions
+                        if variable in TEMP_SENSOR_TYPES]
 
-    async def async_added_to_hass(self):
-        """Register update signal handler."""
-        async def async_update_state():
-            """Update sensor state."""
-            await self.async_update_ha_state(True)
+    for structure, device in nest.smoke_co_alarms():
+        all_sensors += [NestBasicSensor(structure, device, variable)
+                        for variable in conditions
+                        if variable in PROTECT_SENSOR_TYPES]
 
-        async_dispatcher_connect(self.hass, SIGNAL_NEST_UPDATE,
-                                 async_update_state)
+    add_devices(all_sensors, True)
 
 
-class NestBasicSensor(NestSensor):
+class NestBasicSensor(NestSensorDevice):
     """Representation a basic Nest sensor."""
 
     @property
     def state(self):
         """Return the state of the sensor."""
         return self._state
 
+    @property
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return SENSOR_DEVICE_CLASSES.get(self.variable, None)
+
     def update(self):
         """Retrieve latest state."""
         self._unit = SENSOR_UNITS.get(self.variable, None)
 
         if self.variable in VARIABLE_NAME_MAPPING:
             self._state = getattr(self.device,
                                   VARIABLE_NAME_MAPPING[self.variable])
+        elif self.variable in PROTECT_SENSOR_TYPES:
+            # keep backward compatibility
+            self._state = getattr(self.device, self.variable).capitalize()
```

Comment:
State strings are preferably lowercase snakecase. Not sure if it's worth a breaking change though.

Reply:
Those are for Nest Protect. I guess there are lots of automation depends on them.